### PR TITLE
Create meta.yaml for pyocd

### PR DESCRIPTION
**Pull request by rudispr**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by rudispr** [comment]
@conda-forge-admin, please restart ci

**Comment by rudispr** [comment]
@conda-forge-admin, please restart ci

**Comment by rudispr** [comment]
@conda-forge-admin, please restart ci

**Comment by rudispr** [comment]
Hello @conda-forge/help-python  I'm having some trouble with the intelhex dependency. Could you please have a look at the meta.yaml and see if there is anything I might have missed? The builds keep failing during the tests with the error "The 'intelhex<3.0,>=2.0' distribution was not found and is required by the application" for the _pyocd/pyocd-gdbserver --help_ tests or "pyocd 0.0.0 requires intelhex, which is not installed" for the _pip check_ test. I found it odd that the intelhex package seemed to not be installed, so I added the _python -m intelhex --help_ test, and that passes just fine, indicating that the package is available. Do you know what might be the issue? Thanks!

File: recipes/pyocd/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+test:
+  commands:
+    - python -m intelhex --help
+    - pip check
+    - pyocd --help
+    - pyocd-gdbserver --help
+  requires:
+    - pip
+    - intelhex
```
Comment (carterbox) [inline recipes/pyocd/meta.yaml:51]:
```suggestion
test:
  commands:
    - python -m intelhex --help
#    - pip check  # intelhex not configured for pip
    - pyocd --help
    - pyocd-gdbserver --help
  requires:
    - pip
```

File: recipes/pyocd/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/pyocd-{{ version }}.tar.gz
+  sha256: dbc6eb231dc6ada9ff53f493d26fa7a8f6d315d4e641783eceb1dc4a79e240ac
+
+build:
+  skip: true  # [py2k]
+  entry_points:
+    - pyocd = pyocd.__main__:main
+    - pyocd-gdbserver = pyocd.tools.gdb_server:main
+  script: {{ PYTHON }} -m pip install . -vv
+  number: 0
```
Comment (carterbox) [inline recipes/pyocd/meta.yaml:18]:
```suggestion
build:
  noarch: python
  entry_points:
    - pyocd = pyocd.__main__:main
    - pyocd-gdbserver = pyocd.tools.gdb_server:main
  script: {{ PYTHON }} -m pip install . -vv
  number: 0
```

File: recipes/pyocd/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    - libusb-package >=1.0,<2.0
+    - natsort >=8.0.0,<9.0
+    - prettytable >=2.0,<4.0
+    - pyelftools <1.0
+    - pylink-square >=0.11.1,<1.0
+    - python
+    - pyusb >=1.2.1,<2.0
+    - pyyaml >=6.0,<7.0
+    - six >=1.15.0,<2.0
+    - typing-extensions >=4.0,<5.0
```
Comment (carterbox) [inline recipes/pyocd/meta.yaml:41]:
```suggestion
requirements:
  host:
    - pip
    - python >=3.6
  run:
    - capstone >=4.0,<5.0
    - cmsis-pack-manager >=0.4.0,<1.0
    - colorama <1.0
    - dataclasses
    - hidapi >=0.10.1,<1.0
    - intelhex >=2.0,<3.0 
    - intervaltree >=3.0.2,<4.0
    - libusb-package >=1.0,<2.0
    - natsort >=8.0.0,<9.0
    - prettytable >=2.0,<4.0
    - pyelftools <1.0
    - pylink-square >=0.11.1,<1.0
    - python >=3.6
    - pyusb >=1.2.1,<2.0
    - pyyaml >=6.0,<7.0
    - six >=1.15.0,<2.0
    - typing-extensions >=4.0,<5.0
```
https://github.com/pyocd/pyOCD/blob/800abc3dd6f05de890755a542677dabc13296574/setup.cfg#L47

Also, the dataclasses package can be installed on later python versions.

**Review by carterbox — CHANGES_REQUESTED** [review]

**Comment by carterbox** [comment]
It looks like pyocd uses pkg_resources to check its own dependencies at runtime instead of just looking for failed imports. You will have to wait until the intelhex conda package is fixed.

**Comment by carterbox** [comment]
Please ping again when this issue is resolved.

**Comment by rudispr** [comment]
@conda-forge-admin, please restart ci

**Comment by rudispr** [comment]
@carterbox  thanks a lot for the help! It looks like intelhex has been patched and pyocd builds successfully. Can we merge it?

**Pull request merged**